### PR TITLE
add dockerfile to containerize conman

### PR DESCRIPTION
**Pull request by JarettBakerDunn**
Also add git to build environment dockerfile.
Added the docker-devel script to make it easier to use conman inside the container.

### CODE REVIEW

File: docker-devel [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+docker run -it --rm -v $HOME/conman:/home/conman_user/work conman
```
Comment (gassmoeller) [inline docker-devel]: please add a newline at the end of the file (VS code tends to not store with ending newline and github complains about it).

File: docker/conman/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+WORKDIR /home/conman_user
+
+RUN git clone 'https://github.com/geodynamics/conman.git'
+
+WORKDIR /home/conman_user/conman/src
+
+RUN make
+
+WORKDIR /home/conman_user
```
Comment (gassmoeller) [inline docker/conman/Dockerfile]: same as above

Comment (gassmoeller) [inline docker/conman/Dockerfile]: You could simplify this without change of functionality as:
```suggestion

RUN cd src; make
```

**Review by gassmoeller — COMMENTED** [review]
This looks good already, thanks. Two things before we move forward:
1. Can you make sure the created executable is in the PATH environment so that users can start models wherever without knowing the location of the executable?
2. Can you upload the generated docker image to your account already? I would like to test the docker image before merging this PR.

File: docker-devel (hunk 1 above)
Comment (gassmoeller) [inline docker-devel]: What is the purpose of this file?

File: docker/conman/README.md [hunk 3]
```diff
@@ -0,0 +1 @@
+This container hosts a built version of conman.
```
Comment (gassmoeller) [inline docker/conman/README.md:1]: Please include command to run the container here (the line from the docker-devel file above).

**Review by gassmoeller — COMMENTED** [review]
Very nice, almost ready except for my small comments.

However, when I try to execute your container after pulling from docker hub I get the following error message:
```
exec /home/conman_user/conman/conman: exec format error
```
if I run without a command, or
```
exec /bin/bash: exec format error
```
if I start bash explicitly.
This looks as if the image was built for the wrong architecture, but the tag on docker hub says it was build for linux/amd64. How did you create the linux/amd64 image? Does the image run on your system? For comparison your image for `geodynamics/mag` runs fine for me.

**Comment by JarettBakerDunn** [comment]
> Very nice, almost ready except for my small comments.
> 
> However, when I try to execute your container after pulling from docker hub I get the following error message:
> 
> ```
> exec /home/conman_user/conman/conman: exec format error
> ```
> 
> if I run without a command, or
> 
> ```
> exec /bin/bash: exec format error
> ```
> 
> if I start bash explicitly. This looks as if the image was built for the wrong architecture, but the tag on docker hub says it was build for linux/amd64. How did you create the linux/amd64 image? Does the image run on your system? For comparison your image for `geodynamics/mag` runs fine for me.

I built both images the same way using the  --platform=linux/amd64 tag. Both images run on my machine, and both give a warning that the images' platform was built for amd64. I wonder if the conman image won't run for you because I pushed an arm64 image to jbakerdunn/conman before later updating the image with one built for amd64. Maybe some of the layers were not updated? I'll try deleting jbakerdunn/conman and reuploading the conman image.

File: docker/conman/README.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+
+docker run -it --rm -v $HOME/conman:/home/conman_user/work conman
```
Comment (gassmoeller) [inline docker/conman/README.md]:
```suggestion
docker run -it --rm -v $HOME/conman:/home/conman_user/work geodynamics/conman
```

**Comment by gassmoeller** [comment]
This looks good now, thanks!

**Pull request merged**